src/ariel-os-random/src/lib.rs

```diff
@@ -134,17 +129,17 @@ mod csprng {
 
 /// Populates the global RNG from a seed value.
 ///
-/// This is called by Ariel OS's initialization functions.
+/// Only the first call to this function will actually populate the global RNG, following calls
+/// will be no-operations.
```

Comment:
I think it'd be better to panic there too – something is clearly wrong if this happens.